### PR TITLE
Document custom test profile usage with Maven and Gradle

### LINKED ISSUE #3146: Document that running the test suite under a custom test profile is done via
Background info https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!msg/quarkus-dev/NFpDtpRVZ7Y/3G7WmRMJDAAJ

We want to maintain an intuitive experience around using profile configuration:

1. `mvn clean install package -Dquarkus.profile=foo` should build the jar or native image with the `foo` profile but still use the `test` profile when running test

In other words, the profile passed to maven as arguments will not be propagated to the tests, only to the build process. Not doing so leads to confusing experiences and harder to write test suites. This is the current behavior AFAIK.

However, it is useful in some situations to have custom test profiles for example to ensure (part of) the test suite runs for different databases.

To achieve that, we will use the build tool native mechanism (Maven surefire configuration via maven profile, and the equivalent Gradle way). A user would write a surefire config driven by a custom Maven profile that will

1. set the user specific test specificities (maybe even filter the test suite)
2. pass as input parameter to surefire the Quarkus custom (test) profile expected.

We need documentation showing people how to do this.

### PR DESCRIPTION
**Pull request by gwenneg**
Fixes #3146

### CODE REVIEW

**Comment by gwenneg** [comment]
@gsmet: I need to confirm the `NativeImageLauncher` change isn't breaking anything, I'll try to test that ASAP. In the meantime, it would be great if you could start reviewing the PR (especially the doc part).

**Comment by gwenneg** [comment]
The commit message could be changed if you don't want to draw too much attention on the system property.

**Comment by geoand** [comment]
Tests are failing

**Comment by gwenneg** [comment]
@geoand Sorry about that, I forgot to reset the ProfileManager state between tests. It should be better now!

File: docs/src/main/asciidoc/maven-tooling.adoc [hunk 1]
```diff
@@ -485,3 +485,39 @@ Uber-Jar's final name is configurable via a Maven's build settings `finalName` o
...
+  [...]
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-surefire-plugin</artifactId>
+        <version>3.0.0-M3</version>
+        <configuration>
+          <systemPropertyVariables>
+            <quarkus.test.profile>foo</quarkus.test.profile> <1>
```
Comment (gsmet) [inline docs/src/main/asciidoc/maven-tooling.adoc]: We should have an example for failsafe too.

**Review by gsmet — COMMENTED** [review]
Nice work (should I say "as usual"?).

Could you add a Failsafe example? Otherwise, I think it looks good to go.

File: docs/src/main/asciidoc/maven-tooling.adoc (hunk 1 above)
Comment (gwenneg) [reply docs/src/main/asciidoc/maven-tooling.adoc]: You're right, I forgot about it. It's fixed.

File: docs/src/main/asciidoc/maven-tooling.adoc [hunk 2]
```diff
@@ -485,3 +485,51 @@ Uber-Jar's final name is configurable via a Maven's build settings `finalName` o
...
+[source,xml,subs=attributes+]
+----
+<project>
+  [...]
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-surefire-plugin</artifactId>
+        <version>3.0.0-M3</version>
```
Comment (gsmet) [inline docs/src/main/asciidoc/maven-tooling.adoc]: Last minute thought: I'm torn here. I think having the versions is not a very good idea: they will become outdated quickly. So, either we consider people have a plugin management section or we insert the version we have in Quarkus or (and it has my preference) we follow the pattern we have in our generated projects and we use `${surefire-plugin.version}` (we use it also for Failsafe)?

WDYT?

Comment (gwenneg) [reply docs/src/main/asciidoc/maven-tooling.adoc]: I vote for staying consistent with the generated projects. Versions removed :)

**Review by gsmet — APPROVED** [review]
Looks good to me. Thanks!

**Pull request merged**